lib/action-group.ts

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2018 Balena Ltd
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+   http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import * as _ from 'lodash';
+import * as minimatch from 'minimatch';
+import * as path from 'path';
+
+export type Command = string;
+
+export interface StageCopy {
+	source: string;
+	dest: string;
+	sourceStage: number;
+}
+
+export interface LocalCopy {
+	source: string;
+	dest: string;
+}
+
+export interface StageDependentActionGroup {
+	dependentOnStage: true;
+	commands: Command[];
+	workdir: string;
+	copies: StageCopy[];
+	stageDependency: number;
+}
```

Comment:
NIT: `LocalCopy` and `StageDependentActionGroup` could extend the same parent interface.

Reply:
I guess you mean `StageDependentActionGroup` and `LocalDependentActionGroup` ?

Reply:
yes